### PR TITLE
Read CSV as a matrix function instead of pandas.read_csv().as_matrix()

### LINKED ISSUE #197: Unnecessary dependency: Pandas
We should not depend on pandas, it makes Mangaki harder to install and developers less happy because of the long compile time.

Actually, we are using pandas _only for reading CSV_, which is (as @jilljenn said): **stupid** :smile: !

- Relevant lines: https://github.com/mangaki/mangaki/blob/master/mangaki/mangaki/management/commands/compare.py#L41 and https://github.com/mangaki/mangaki/blob/master/mangaki/mangaki/management/commands/compare.py#L41

I will come up with a simple replacement for the CSV loading.

EDIT: Harder than expected.
Reading the CSV is fine, but transforming it into a proper NumPy Array with the good data types is not trivial.

For ratings: we have Integer, Integer, String.
For works: we have Integer, String.

Now, we could just use a `StringIO` and do `np.fromfile` or `np.genfromtxt` with the proper dtypes.
Unfortunately, it looks like these functions are bugged in Python 3.5 and will throw a `TypeError: Can't convert 'bytes' object to str implicitly'`.

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Thanks to @Elarnon, here is a similar `read_csv…as_matrix` replacement which is not too annoying at the same time.

We could close #197 with this, I tested multiple times the command and it works (just got a bug with divide-by-zero on the SVD algorithm, due to NaN values most likely).

I invite the Research Team to try this out and _compare_ if the RMSE / shapes are relevant vs the old version so we don't have any regression. (/ping @jilljenn @ejls @Elarnon)

### CODE REVIEW

File: mangaki/mangaki/utils/svd.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def make_matrix(self, X, y):
             matrix[user][work] = rating
         means = np.zeros((self.nb_users,))
         for i in range(self.nb_users):
-            means[i] = np.sum(matrix[i]) / np.sum(matrix[i] != 0)
+            means[i] = np.nanmean(matrix[i])
```
Comment (Elarnon) [inline mangaki/mangaki/utils/svd.py]: This is not the same thing; `means` should contain the mean of *nonzero* elements of `matrix[i]` while this computes the mean of all elements.

File: mangaki/mangaki/management/commands/compare.py [hunk 2]
```diff
@@ -10,11 +10,19 @@
 from collections import Counter
 import numpy as np
 import random
-import pandas
+import csv
 # import matplotlib.pyplot as plt
```
Comment (Elarnon) [inline mangaki/mangaki/management/commands/compare.py]: While we are here, can we remove this ?

File: mangaki/mangaki/utils/svd.py (hunk 1 above)
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/svd.py]: Right, indeed.

File: mangaki/mangaki/management/commands/compare.py (hunk 2 above)
Comment (RaitoBezarius) [inline mangaki/mangaki/management/commands/compare.py]: Sure!

File: mangaki/mangaki/utils/svd.py (hunk 1 above)
Comment (RaitoBezarius) [inline mangaki/mangaki/utils/svd.py]: To me: `np.sum(matrix[i]) / np.count_nonzero(matrix[i])` is fine as a replacement (assuming `np.count_nonzero(matrix[i]) != 0`), am I right?

File: mangaki/mangaki/utils/svd.py [hunk 3]
```diff
@@ -52,9 +52,14 @@ def make_matrix(self, X, y):
             matrix[user][work] = rating
         means = np.zeros((self.nb_users,))
         for i in range(self.nb_users):
-            means[i] = np.sum(matrix[i]) / np.sum(matrix[i] != 0)
+            count_nonzero = np.count_nonzero(matrix[i])
+
+            if count_nonzero != 0:
```
Comment (Elarnon) [inline mangaki/mangaki/utils/svd.py]: This case is handled by the `if np.isnan(means[i])` below.

**Comment by RaitoBezarius** [comment]
OK, bon je pense qu'on peut garder `pandas` en dépendance, tu as totalement raison @Elarnon — j'abandonne ! :D 
Je ferme.

**Pull request closed without merging**